components/higher-order/with-global-events/test/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { mount } from 'enzyme';
+
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import withGlobalEvents from '../';
+import Listener from '../listener';
+
+jest.mock( '../listener', () => {
+	const ActualListener = require.requireActual( '../listener' ).default;
+
+	return class extends ActualListener {
+		constructor() {
+			super( ...arguments );
+
+			this.constructor._instance = this;
+
+			jest.spyOn( this, 'add' );
+			jest.spyOn( this, 'remove' );
+		}
+	};
+} );
+
+describe( 'withGlobalEvents', () => {
+	let wrapper;
+
+	class OriginalComponent extends Component {
+		handleResize( event ) {
+			this.props.onResize( event );
+		}
+
+		render() {
+			return <div>{ this.props.children }</div>;
+		}
+	}
+
+	beforeAll( () => {
+		jest.spyOn( OriginalComponent.prototype, 'handleResize' );
+	} );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	afterEach( () => {
+		if ( wrapper ) {
+			wrapper.unmount();
+			wrapper = null;
+		}
+	} );
+
+	function mountEnhancedComponent( props ) {
+		const EnhancedComponent = withGlobalEvents( {
+			resize: 'handleResize',
+		} )( OriginalComponent );
+
+		wrapper = mount( <EnhancedComponent { ...props }>Hello</EnhancedComponent> );
```

Comment:
I started to think if we should introduce our own `mount` or further enrich `enzyme` to call `unmount` behind the scenes when the test finishes.

Reply:
```js
let wrapper;
export const mount = ( ...params ) => {
	wrapper = originalMount( ...params );
	return wrapper;
};

afterEach( () => {
	if ( wrapper ) {
		wrapper.unmount();
		wrapper = null;
	}
} );
```

Reply:
>I started to think if we should introduce our own `mount` or further enrich `enzyme` to call `unmount` behind the scenes when the test finishes.

I'd like this 👍

Reply:
I will check it next week. `jest.mount( element )` or `jest.shallow( element )` might also save some keystrokes :)